torch/csrc/distributed/c10d/reducer.hpp

```diff
@@ -41,14 +57,45 @@ class TORCH_API Timer {
     kBackwardCommEnd,
   };
 
-  // Record the current event, i.e., mark it as having occurred now.
-  virtual void record(Event event) = 0;
+  // Record the current event, i.e., mark it as having occurred now. Default
+  // CPU implementation.
+  virtual void record(Event event) {
+    getTime(event) = current_time_in_nanos();
+  }
 
   // Return the difference between when two events occurred, in nanoseconds.
   // Or nullopt if one of them hasn't been recorded.
   virtual c10::optional<int64_t> measureDifference(Event start, Event end) = 0;
 
   virtual ~Timer() = default;
+
+  // Return host-side timestamp, or nullopt if it has not yet been recorded.
+  c10::optional<int64_t> getTimestamp(Event event) {
+    auto time = getTime(event);
+    if (time == kUnsetTime) {
+      return c10::nullopt;
+    } else {
+      return time;
+    }
+  }
+
+  // Return host-side time member variable corresponding to the given event.
```

Comment:
Since it's specific to "host-side", will it make more sense to move it back to `CPUTime` class? Or do you want to update the comment?

Reply:
See above comment regarding this